modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/enums/CredentialType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.credential.enums;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * The type of the credential.
+ */
+public enum CredentialType {
+    INPUT("input"),

```

Comment:
would it make sense to have HTTP_AUTHENTICATION, APKI_KEY and OAUTH2 as types?

Just to avoid the generic 'INPUT' which is abstract.